src/main/frontend/src/app/pages/release-graph/release-node.service.ts

```diff
@@ -889,36 +949,127 @@ export class ReleaseNodeService {
-          node.position.x = previousNode.position.x + currentMinSpacing;
-        }
-      }
+      if (nodeVersion.patch > latestVersion.patch) return node;
+      return latest;
+    });
+  }
+
+  private findLatestNightly(nodes: ReleaseNode[]): ReleaseNode | null {
+    const nightlies = nodes.filter((n) => this.isNightlyRelease(n.label));
+    if (nightlies.length === 0) return null;
+
+    return nightlies.reduce((latest, node) => {
+      return node.publishedAt > latest.publishedAt ? node : latest;
+    });
+  }
+
+  private findLastNonNightlyRelease(nodes: ReleaseNode[]): ReleaseNode | null {
+    const nonNightly = nodes.filter((n) => !this.isNightlyRelease(n.label));
+    if (nonNightly.length === 0) return null;
+
+    return nonNightly.reduce((latest, node) => {
+      return node.publishedAt > latest.publishedAt ? node : latest;
+    });
+  }
+
+  private isBranchSupported(nodes: ReleaseNode[]): boolean {
+    const firstRelease = nodes.find((n) => !this.isNightlyRelease(n.label));
+    if (!firstRelease) return false;
+
+    const versionInfo = this.getVersionInfo(firstRelease);
+    if (!versionInfo) return false;
+
+    const supportDates = this.getSupportEndDates(firstRelease);
+    if (!supportDates) return false;
+
+    return new Date() <= supportDates.securitySupportEnd;
+  }
+
+  private isEOLBranch(nodes: ReleaseNode[]): boolean {
+    const hasNightly = nodes.some((n) => this.isNightlyRelease(n.label));
+    if (!hasNightly) return false;
+
+    return this.isBranchUnsupported(nodes);
+  }
+
+  private isBranchUnsupported(nodes: ReleaseNode[]): boolean {
+    const firstRelease = nodes.find((n) => !this.isNightlyRelease(n.label));
+    if (!firstRelease) return true;
+
+    const supportDates = this.getSupportEndDates(firstRelease);
+    if (!supportDates) return true;
 
-      spacedFinalNodes.push(node);
+    return new Date() > supportDates.securitySupportEnd;
+  }
+
+  private findLatestLTS(nodes: ReleaseNode[]): ReleaseNode | null {
+    const majorVersions = nodes.filter((n) => {
+      const versionInfo = this.getVersionInfo(n);
+      return versionInfo?.type === 'major';
+    });
+
+    if (majorVersions.length === 0) return null;
+
+    return majorVersions.reduce((latest, node) => {
+      const latestVersion = this.getVersionInfo(latest);
+      const nodeVersion = this.getVersionInfo(node);
+
+      if (!latestVersion || !nodeVersion) return latest;
+
+      if (nodeVersion.major > latestVersion.major) return node;
+      return latest;
+    });
+  }
+
+  private isUnsupported(release: ReleaseNode): boolean {
+    if (release.label.toLowerCase().includes(ReleaseNodeService.GITHUB_SNAPSHOT_DISPLAY)) {
+      return false;
+    }
+
+    const supportDates = this.getSupportEndDates(release);
+    if (!supportDates) return true;
+
+    return new Date() > supportDates.securitySupportEnd;
+  }
+
+  private getSupportEndDates(release: ReleaseNode): { fullSupportEnd: Date; securitySupportEnd: Date } | null {
```

Comment:
```
{ fullSupportEnd: Date; securitySupportEnd: Date }
```
I've seen you use this type in multiple places, maybe you could make it an actual reusable type?
```ts
type SupportedDates = { fullSupportEnd: Date; securitySupportEnd: Date }
```